packages/arb-token-bridge-ui/src/hooks/useCCTP.ts

```diff
@@ -0,0 +1,324 @@
+    USDCContractAddress: '0x07865c6e87b9f70255377e024ace6630c1eaa37f',
+    messengerTransmitterContractAddress:
+      '0x109bc137cb64eab7c0b1dddd1edf341467dc2d35',
+    attestation_API_URL: 'https://iris-api-sandbox.circle.com'
+  },
+  [ChainId.ArbitrumOne]: {
+    tokenMessengerContractAddress: '0x19330d10D9Cc8751218eaf51E8885D058642E08A',
+    targetChainDomain: 0,
+    targetChainId: ChainId.Mainnet,
+    USDCContractAddress: '0xaf88d065e77c8cc2239327c5edb3a432268e5831',
+    messengerTransmitterContractAddress:
+      '0x0a992d191deec32afe36203ad87d7d289a738f81',
+    attestation_API_URL: 'https://iris-api.circle.com'
+  },
+  [ChainId.ArbitrumGoerli]: {
+    tokenMessengerContractAddress: '0x12dcfd3fe2e9eac2859fd1ed86d2ab8c5a2f9352',
+    targetChainDomain: 0,
+    targetChainId: ChainId.Goerli,
+    USDCContractAddress: '0xfd064a18f3bf249cf1f87fc203e90d8f650f2d63',
+    messengerTransmitterContractAddress:
+      '0x26413e8157cd32011e726065a5462e97dd4d03d9',
+    attestation_API_URL: 'https://iris-api-sandbox.circle.com'
+  }
+}
+
+function getL1AddressFromAddress(address: string) {
+  switch (address) {
+    case CommonAddress.ArbitrumGoerli.USDC:
+      return CommonAddress.Goerli.USDC
+    case CommonAddress.ArbitrumGoerli['USDC.e']:
+      return CommonAddress.Goerli.USDC
+    case CommonAddress.ArbitrumOne.USDC:
+      return CommonAddress.Mainnet.USDC
+    case CommonAddress.ArbitrumOne['USDC.e']:
+      return CommonAddress.Mainnet.USDC
+    default:
+      return CommonAddress.Mainnet.USDC
+  }
+}
+
+type AttestationResponse =
+  | {
+      attestation: `0x${string}`
+      status: 'complete'
+    }
+  | {
+      attestation: null
+      status: 'pending_confirmations'
+    }
+
+function getContracts(chainId: ChainIdKeys | undefined) {
+  if (!chainId) {
+    return contracts[ChainId.Mainnet]
+  }
+  return contracts[chainId]
+}
+
+export function useCCTP({
+  chainId,
+  walletAddress
+}: {
+  chainId?: ChainIdKeys
+  walletAddress?: `0x${string}` | string
+}) {
+  const {
+    tokenMessengerContractAddress,
+    targetChainDomain,
+    targetChainId,
+    attestation_API_URL,
+    USDCContractAddress,
+    messengerTransmitterContractAddress
+  } = getContracts(chainId)
+  const { l1, l2 } = useNetworksAndSigners()
+  const {
+    erc20: [, updateErc20L1Balance]
+  } = useBalance({
+    provider: l1.provider,
+    walletAddress
+  })
+  const {
+    erc20: [, updateErc20L2Balance]
+  } = useBalance({
+    provider: l2.provider,
+    walletAddress
+  })
+
+  const destinationAddress = useMemo(() => {
+    if (!walletAddress) return
+
+    // CCTP uses 32 bytes addresses, while EVEM uses 20 bytes addresses
+    return utils.hexlify(utils.zeroPad(walletAddress, 32)) as `0x${string}`
+  }, [walletAddress])
+
+  const { writeAsync: depositForBurn } = useContractWrite({
+    address: tokenMessengerContractAddress,
+    abi: tokenMessengerAbi,
+    functionName: 'depositForBurn',
+    mode: 'recklesslyUnprepared'
```

Comment:
Is there some other way we could do this, or is this the only way? I would also add some additional context via comments and links to the documentation, potentially even some todos on how/when can this be fixed

Reply:
Moved to `@wagmi/actions`